pkg/console/util.go

```diff
@@ -982,3 +986,78 @@ func generateEnvAndConfig(g *gocui.Gui, hvstConfig *config.HarvesterConfig) ([]s
 	env = append(env, fmt.Sprintf("HARVESTER_STREAMDISK_CLOUDINIT_URL=%s", userDataURL))
 	return env, elementalConfig, nil
 }
+
+// internal objects to parse lsblk output
+type BlockDevices struct {
+	Disks []Device `json:"blockdevices"`
+}
+
+type Device struct {
+	Name     string   `json:"name"`
+	Size     string   `json:"size"`
+	DiskType string   `json:"type"`
+	WWN      string   `json:"wwn,omitempty"`
+	Serial   string   `json:"serial,omitempty"`
+	Children []Device `json:"children,omitempty"`
+}
+
+func generateDiskEntry(d Device) string {
+	return fmt.Sprintf("%s %s", d.Name, d.Size)
+}
+
+const (
+	diskType = "disk"
+)
+
+// identifyUniqueDisks parses the json output of lsblk and identifies
+// unique disks by comparing their child elements
```

Comment:
There's a few comments in this function referring to children, which I think don't apply anymore (looks like it's going by Serial and WWN)

Reply:
this has now been fixed to better reflect how we are using serial number and/or wwn to identify unique disks